account_invoice_validation_purchase_type/tests/test_account_invoice_validation_purchase_type.py

```diff
@@ -0,0 +1,201 @@
+# Copyright 2018 ACSONE SA/NV (<http://acsone.eu>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo.fields import Command
+from odoo.tests.common import TransactionCase
+
+
+class TestAccountInvoice(TransactionCase):
+    """
+    Tests for account.move
+    """
+
+    def setUp(self):
+        super().setUp()
+        self.env = self.env(context=dict(self.env.context, tracking_disable=True))
+        self.precision = self.env["decimal.precision"].precision_get(
+            "Product Unit of Measure"
+        )
+        self.AccountMove = self.env["account.move"]
+        self.AccountAccount = self.env["account.account"]
+        self.Product = self.env["product.product"]
+        self.supplier = self.env.ref("base.res_partner_3")
+        self._prepare_tests()
+
+    def _prepare_tests(self):
+        """
+        Prepare data for the next tests
+        :return: bool
+        """
+        Product = self.Product
+        AccountMove = self.AccountMove
+        AccountAccount = self.AccountAccount
+        AccountTax = self.env["account.tax"]
+        supplier = self.supplier
+        account_receivable = AccountAccount.create(
+            {
+                "code": "200000bis",
+                "name": "Sale unit test",
+                "account_type": "income",
+                "reconcile": True,
+            }
+        )
+        account_payable = AccountAccount.create(
+            {
+                "code": "220000bis",
+                "name": "Expenses unit test",
+                "account_type": "expense",
+                "reconcile": True,
+            }
+        )
+        tax_p1 = AccountTax.create(
+            {
+                "name": "percent_tax",
+                "amount_type": "percent",
+                "amount": 10.0,
+            }
+        )
+        tax_p2 = AccountTax.create(
+            {
+                "name": "percent_tax",
+                "amount_type": "percent",
+                "amount": 10.0,
+            }
+        )
+        self.product1 = Product.create(
+            {
+                "name": "Product purchase type - Unit test 1",
+                "purchase_type": True,
+                "list_price": 500,
+                "property_account_expense_id": account_payable.id,
+                "taxes_id": tax_p1.ids,
```

Comment:
You should use `Command.xxx(...)`. Giving a list of id works only because it's a test.